pkg/scheduler/schedule_one.go

```diff
@@ -441,14 +441,13 @@ func (sched *Scheduler) schedulePod(ctx context.Context, fwk framework.Framework
 // filter plugins and filter extenders.
 func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.Framework, state *framework.CycleState, pod *v1.Pod) ([]*framework.NodeInfo, framework.Diagnosis, error) {
 	logger := klog.FromContext(ctx)
+	diagnosis := framework.Diagnosis{
+		NodeToStatusMap: make(framework.NodeToStatusMap),
+	}
 
 	allNodes, err := sched.nodeInfoSnapshot.NodeInfos().List()
 	if err != nil {
-		return nil, framework.Diagnosis{}, err
-	}
-
-	diagnosis := framework.Diagnosis{
-		NodeToStatusMap: make(framework.NodeToStatusMap, len(allNodes)),
+		return nil, diagnosis, err
 	}
 	// Run "prefilter" plugins.
 	preRes, s := fwk.RunPreFilterPlugins(ctx, state, pod)
```

Comment:
I can't comment on an unchanged line.
So based on our implementation, in L460, we should only set status for all nodes if the status returned by `fwk.RunPreFilterPlugins(ctx, state, pod)` has `Unschedulable` code, right? Which can only happen in a scheduler with specific out-of-tree plugins. And can reduce the candidates during preempetion.

Reply:
Yes, this sounds right to me. However, I omited it from this change to keep the diff small, since updating the tests produces a 30 line (+7, -23) diff.  Then, the intention is to clean it up in a PR which won't be cherry-picked.

Do you prefer I include it in this change, or do you think the original plan makes sense?